contracts/src/bridge/Outbox.sol

```diff
@@ -45,28 +62,51 @@ contract Outbox is DelegateCallAware, IOutbox {
     /// When the return value is zero, that means this is a system message
     /// @dev the l2ToL1Sender behaves as the tx.origin, the msg.sender should be validated to protect against reentrancies
     function l2ToL1Sender() external view override returns (address) {
-        return context.sender;
+        address sender = context.sender;
+        // we don't return the default context value to avoid a breaking change in the API
+        if(sender == getDefaultContext().sender) return address(0);
```

Comment:
I'm wondering how well solidity optimizes this. It might be better to to define constants for the default value of each field and then implement `getDefaultContext` using those constants